tests/unit/test_modelling.py

```diff
@@ -278,6 +278,85 @@ def test_from_array_not_finite_error(self):
 
         self.assertEqual(msg, str(cm.exception))
 
+    def test_sequence_from_array(self):
+        """Test that a sequence of inputs can be generated from a
+        sequence of arrays."""
+
+        _inputs = np.array(
+            [
+                [1, 2, 3],
+                [4, 5, 6],
+            ]
+        )
+        expected = (Input(1, 2, 3), Input(4, 5, 6))
+        self.assertEqual(expected, Input.sequence_from_array(_inputs))
+
+    def test_sequence_from_array_lists_tuples(self):
+        """Test that a sequence of arrays is taken in and still correctly functions."""
+
+        # From a list
+        _inputs1 = [np.array([1, 2, 3]), np.array([4, 5, 6])]
+        expected1 = (Input(1, 2, 3), Input(4, 5, 6))
+        self.assertEqual(expected1, Input.sequence_from_array(_inputs1))
+
+        # From a tuple
+        _inputs2 = (np.array([1, 2, 3]), np.array([4, 5, 6]))
+        expected2 = (Input(1, 2, 3), Input(4, 5, 6))
+        self.assertEqual(expected2, Input.sequence_from_array(_inputs2))
+
+    def test_sequence_from_array_type_error(self):
+        """Ensure that a TypeError is raised if a non-sequence type is passed."""
+
+        _inputs = 32
+        msg = f"Expected 'inputs' to be of type Sequence of np.ndarray or 2D np.ndarray, but received {type(_inputs)} instead."
+        with self.assertRaisesRegex(TypeError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
+
+    def test_sequence_from_array_wrong_dimension(self):
+        """Test a ValueError is raised if the wrong dimensional array is passed"""
+
+        # 3D case
+        _inputs = np.array(
+            [
+                [[1, 2], [3, 4], [5, 6], [7, 8]],
+                [[9, 10], [11, 12], [13, 14], [15, 16]],
+                [[17, 18], [19, 20], [21, 22], [23, 24]],
+            ]
+        )
+        msg = f"Expected np.array of dimension 2, but received {_inputs.ndim} dimensions."
+
+        with self.assertRaisesRegex(ValueError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
+
+        # 1D case
+        _inputs = np.array([1, 2, 3])
+        with self.assertRaisesRegex(ValueError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
+
+        # Empty array case
+        _inputs = np.array([])
+        with self.assertRaisesRegex(ValueError, exact(msg)):
+            _ = Input.sequence_from_array(_inputs)
```

Comment:
Need to redefine dimensions of `_inputs`.

```suggestion
        # 1D case
        _inputs = np.array([1, 2, 3])
        msg = f"Expected np.array of dimension 2, but received {_inputs.ndim} dimensions."
        with self.assertRaisesRegex(ValueError, exact(msg)):
            _ = Input.sequence_from_array(_inputs)

        # Empty array case
        _inputs = np.array([])
        msg = f"Expected np.array of dimension 2, but received {_inputs.ndim} dimensions."
        with self.assertRaisesRegex(ValueError, exact(msg)):
            _ = Input.sequence_from_array(_inputs)
```

Reply:
Ah, you spotted my deliberate mistake!